mh/quest_utils.py

```diff
@@ -1325,6 +1325,9 @@ def read_quest_sm_data(fname):
         data = f.read()
     return bytearray.fromhex(data.replace(" ","").replace("\n",""))
 
+def get_quest_name(quest):
+    return quest[0:40].decode('ascii').rstrip('\x00')
```

Comment:
This should return `bytes` so we don't have to handle the encoding. There are other cleaner ways to get a null-terminated string from a buffer of bytes (for instance, using the `ctypes` module).

If the string is guaranteed to be null terminated:
```python
return ctypes.create_string_buffer(quest).value
```
Otherwise, if it has a fixed size and isn't guaranteed to be null terminated:
```python
return ctypes.create_string_buffer(quest[:40]).value
```

There is another way involving slices and `bytes.find` but it will be cumbersome if you have to deal with cases where the string isn't null-terminated.

Reply:
Changed